livekit-agents/livekit/agents/voice_assistant/voice_assistant.py

```diff
@@ -145,6 +160,9 @@ def __init__(
             transcription: Options for assistant transcription.
             will_synthesize_assistant_reply: Callback called when the assistant is about to synthesize a reply.
```

Comment:
nit. these events tend to confuse me a bit. since we are adding a new one, I wonder if there's a better way to handle these.
and should we use callbacks rather than events? 

i.e. assistant.llm_preprocess_cb=myfunc and assistant.llm_postprocess_cb

where both functions can manipulate chat context, and postprocess could also determine what goes into TTS